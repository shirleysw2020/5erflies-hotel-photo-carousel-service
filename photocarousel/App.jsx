import React, { useState, useEffect } from 'react';
import styled ,{ keyframes } from 'styled-components';
import { ThemeProvider } from 'styled-components';
import axios from 'axios';
import Header from './Header.jsx';
import Gallery from './Gallery.jsx';
import Modal from './Modal.jsx';

const Photocarousel = () => {
  const [listing, setListing] = useState([]);
  const [clickedPhotoIdx, setClickedPhoto] = useState(1)
  const [showModal, setShowModal] = useState(null);
  const openModal = (index) => {
    if (typeof index === 'number') {
      setClickedPhoto(index);
    }
    setShowModal(true);
  }
  const closeModal = () => setShowModal(false);
  // console.log('clientb', listing);

  useEffect(() => { // a hooks version of componentDidMount
    const queryID = window.location.search;
    // console.log('logged: window.location (url parameter) = ?id=10', queryID)
    const url = '/listing' + queryID;
    // const url = '/listing' + '/' + queryID;
    axios.get(url)
    .then((res) => {
      // console.log(res.data, "got this data from db");
      setListing(res.data);
    })
    .catch((err) => {
      console.log('client: failed getting db', err);
    })
  }, []);


  if (listing.length == 0) {return null;}
  return (
    <AppWrapper>
      <MainWrapper show={showModal}>
        <Header list={listing[0]}/>
        <Gallery
          list={listing[0]}
          openModal={openModal}
          closeModal={closeModal}
          show={showModal}
        />
      </MainWrapper>
      <Slider show={showModal}>
        <Modal
          list={listing[0]}
          closeModal={closeModal}
          clickedPhotoIdx={clickedPhotoIdx}
          show={showModal}
        />
      </Slider>
    </AppWrapper>
  )
}

const theme = {
  background: '#aaa',
  color: '#24292e'
};

const AppWrapper = styled.div`
  position: relative;
`;

const MainWrapper = styled.div`
  width: 60% ;
  margin: 0 auto;
  margin-top: ${props => props.show ? '0px' : '40px'};
  font-family: sans-serif, Roboto;
  opacity: ${props => props.show ? 0 : 1};
`;

const slideup = keyframes`
  0% {
    /* position relative to itself */
    transform: translateY(0px);
    opacity: 1;
  }
  100% {
    transform: translateY(-450px);
    opacity: 1;
  }
`;

const slidedown = keyframes`
  0% {
    /* if position of itself is fixed, then relative to window, if absolute, then relative to itself */
    transform: translateY(-450px);
    opacity: 1;
  }
  100% {
    transform: translateY(0);
    opacity: 0;
  }
`;

const Slider = styled.div`
/* null to prevent a slidedown / slide up without clicking button */
  display: ${props => props.show ? 'block' : 'none'};
  animation: ${props => props.show === null ? 'none' : props.show ? slideup : slidedown};
  animation-duration: 0.6s;
  animation-fill-mode: forwards;
  opacity: 0;
  width: 100vw;
  height: 100vh;
  z-index: ${props => props.show ? '0' : '-10'};
  display: inline-block;
  position: absolute; //for overlap the gallery
  top: 450px;
`

export default Photocarousel;
