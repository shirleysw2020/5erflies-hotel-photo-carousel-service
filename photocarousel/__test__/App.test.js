import React from 'react';
import { shallow, mount } from 'enzyme';
import { create } from "react-test-renderer";
import renderer from 'react-test-renderer';
import App from '../App.jsx';


describe('First React component test with Enzyme', () => {
   it('renders without crashing', () => {
      shallow(<App />);
    });

    it('should have the Gallery component', () => {
       const wrapper = shallow(<App />)
       expect(wrapper.find('Gallery')).toBeDefined();
    });

    it("Matches the snapshot and return null since App doesn't take in props", () => {
      const tree = renderer.create(<App />).toJSON();
      expect(tree).toMatchSnapshot();
     });
});





// it('the show property should be true with a click of button', () => {
//    const wrapper = mount(<App />);
//    wrapper.find('input[type="password"]').simulate('click', {
//      target: {
//        value: 'validpassword',
//      },
//    });
//    expect(wrapper.find('input[type="button"]').prop('disabled')).toBeFalsy();
//  });

// describe('debugging', () => {
//    it ('debugging App', () => {
//    const shallowWrapper = shallow(<App />);
//    const mountWrapper = mount(<App />);

//    console.log('shallow app', shallowWrapper.debug());
//    console.log('mount app', mountWrapper.debug());
//    })
// })

// describe('just test', () => {
//    it ('calls useEffect at elast once', () => {
//       spy(App.prototype, 'useEffect')
//       const wrapper = mount(<App />);
//       expect(App.prototype.useEffect.calledOnce.toEqual(true));
//    });
// });
// describe('should call openModal method and set state of show', () => {
//    it ('set state of show to be true', () => {
//       const wrapper = mount(<App />);
//       expect(wrapper.prop('show').toBe(true));
//       const openModalInstance = wrapper.instance();
//       openModalInstance.openModal();
//    })
// });

// describe('should call openModal method and set state of show', () => {
//    it ('set state of show to be true', () => {
//       const wrapper = shallow(<App />);
//       expect(wrapper.state('show').toBe(true));
//       const openModalInstance = wrapper.instance();
//       openModalInstance.openModal();
//    })
// });

// const wrapper = shallow(<App />);
// expect(wrapper.find({ prop: 'listing' })).to.have.lengthOf(3);

// describe('<App />', () => {
//    const addCount = () => setphotoCount(photoCount++);
//    it('renders one <Header /> component', () => {
//      const wrapper = shallow(<App/>);
//      expect(wrapper.children()).to.have.lengthOf(3);
//    });
//  });

// A shallow wrapper isn't the component, it's what the component renders. the wrapper you get is around the div it renders - which has no instance

//mount renders a component to its extreme leaf nodes whereas shallow as the name suggests does a shallow render i.e. renders just the component and not its children.