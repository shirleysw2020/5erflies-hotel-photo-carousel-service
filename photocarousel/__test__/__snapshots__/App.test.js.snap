// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First React component test with Enzyme Matches the snapshot and return null since App doesn't take in props 1`] = `null`;
